lbclient.go

```diff
@@ -91,9 +94,36 @@ func (cc *LBClient) init() {
 	}
 }
 
+// AddClient adds a new client to the balanced clients
+// returns the new total number of clients
+func (cc *LBClient) AddClient(c BalancingClient) int {
+	cc.mu.Lock()
+	cc.cs = append(cc.cs, &lbClient{
+		c:           c,
+		healthCheck: cc.HealthCheck,
+	})
+	cc.mu.Unlock()
+	return len(cc.cs)
+}
+
+// RemoveClients removes clients using the provided callback
+// if rc returns true, the passed client will be removed
+// returns the new total number of clients
+func (cc *LBClient) RemoveClients(rc func(BalancingClient) bool) int {
+	cc.mu.Lock()
+	for idx, cs := range cc.cs {
+		if rc(cs.c) {
+			cc.cs = append(cc.cs[:idx], cc.cs[idx+1])
```

Comment:
This is not a correct way to filter a slice. See: https://go.dev/play/p/9y9DyCJ_xDD

Reply:
ah, my bad. addressed